### PR TITLE
added votes and add or minus votes

### PR DESCRIPTION
**Pull request by SyedaKausar**

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -12,15 +12,23 @@ export const getArticles = (topic) => {
...
   });
 };
+export const patchVote = (id, vote) => {
+  return articlesApi
+    .patch(`/articles/${id}`, { inc_votes: vote })
+    .then((res) => {
+      return res.data;
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (popatre) [inline src/api.js]: remove log please

File: src/components/SingleArticle.js [hunk 2]
```diff
@@ -16,8 +19,29 @@ const SingleArticle = () => {
...
-  }, []);
+  }, [article_id]);
+  function addVote() {
+    if (!isClicked) {
+      setCount((currCount) => {
+        return currCount + 1;
+      });
+
+      patchVote(article_id, 1).then((res) => {
+        setIsClicked(true);
```
Comment (popatre) [inline src/components/SingleArticle.js]: Problem with this is that if you clikc the button multiple times quickly, it lets you upvote more than once. 
if it was a slow connection, you'd be able to click it loads before it gets disabled.

Comment (popatre) [reply src/components/SingleArticle.js]: Also, if the api request failed, the button would never be disabled and you'd be able to click it forever. On the client side, would look like its working.

File: src/components/SingleArticle.js [hunk 3]
```diff
@@ -16,8 +19,29 @@ const SingleArticle = () => {
...
+      });
+    }
+  }
```
Comment (popatre) [inline src/components/SingleArticle.js:39]: Need some handling for then when the api request doesnt work, i.e. reseting the count back to what it was before the click.

**Review by popatre — CHANGES_REQUESTED** [review]

**Review by leekli — APPROVED** [review]
Can merge

**Pull request merged**